Hotpool v2 Imlementation

utopiabound opened the pull request:
Core logic to create Hotpool v2.

Fixes #1464

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2241)
<!-- Reviewable:end -->

utopiabound commented:
This is dependent on:
* #2178

jgrund commented on chroma_core/models/filesystem.py:
```diff
@@ -109,6 +109,20 @@ def get_servers(self):
+            if name not in groups:
+                groups[name] = hosts
+
+        return groups.values()
```
This fn isn't accurate as it doesn't account for bans.

jgrund commented on chroma_api/hotpool.py:
```diff
@@ -0,0 +1,96 @@
+# Use of this source code is governed by a MIT-style
+# license that can be found in the LICENSE file.
+
+from collections import defaultdict
```
The API should be part of graphql instead.

jgrund commented on chroma_core/services/job_scheduler/job_scheduler.py:
```diff
@@ -1992,3 +1999,158 @@ def remove_task(self, task_id):
         self.progress.advance()
         return task.id, command_id
+
+    def create_hotpool(self, hotpool_data):
```
This should be a graphql endpoint.

jgrund replied:
Unresolving as current patch still contains change.

jgrund reviewed (changes requested):
There are a few things in here that should leverage graphql instead.

jgrund reviewed (changes requested):
Python hotpool api still needs removal

utopiabound committed:
Add Hotpool v2 support

Create lamigo/lpurge models
Add django migration
Add hpv2 stuff to ObjectCache
Add IML CLI commands
Add graphql hotpool create/remove

Signed-off-by: Nathaniel Clark <nclark@whamcloud.com>

jgrund commented on iml-api/src/graphql/mod.rs:
```diff
@@ -250,6 +272,49 @@ impl QueryRoot {
+        offset(description = "Offset into items, defaults to 0"),
+        dir(description = "Sort direction, defaults to asc"),
+    ))]
+    /// Fetch the list of known targets
```
Update docs to describe params

jgrund commented on iml-wire-types/src/lib.rs:
```diff
@@ -1997,6 +1997,37 @@ pub struct FidItem {
+    /// Hot OstPool
+    pub hot_id: i32,
+    /// Cold OstPool
+    pub cold_id: i32,
```
Update to `cold_pool_name`

jgrund commented on iml-wire-types/src/lib.rs:
```diff
@@ -1997,6 +1997,37 @@ pub struct FidItem {
+    pub freehi: Option<i32>,
+    pub freelo: Option<i32>,
+    /// Hot OstPool
+    pub hot_id: i32,
```
Update to `hot_pool_name`

utopiabound committed:
Fixup: QL to stratagem, pool names in config struct

move hotpool QL to stratagem.rs
Use pool names in HotpoolConfiguration struct.


Signed-off-by: Nathaniel Clark <nclark@whamcloud.com>

utopiabound committed:
FIXUP: Account for changes in HotpoolConfiguration

Signed-off-by: Nathaniel Clark <nclark@whamcloud.com>
